assets/js/modules/analytics/util/report-date-range-args.test.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * Tests for reporting date range utilities.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { generateDateRangeArgs } from './report-date-range-args';
+
+describe( 'Analytics reporting date range arguments', () => {
+	describe( 'generateDateRangeArgs', () => {
+		it( 'should throw if `startDate` or `endDate` arguments are not provided', () => {
+			try {
+				generateDateRangeArgs( { startDate: '2020-12-18' } );
+			} catch ( error ) {
+				expect( error.message ).toEqual( 'A valid endDate is required.' );
+			}
+
+			try {
+				generateDateRangeArgs( { endDate: '2021-01-14' } );
+			} catch ( error ) {
+				expect( error.message ).toEqual( 'A valid startDate is required.' );
+			}
```

Comment:
Please use [`.toThrow`](https://jestjs.io/docs/en/expect#tothrowerror) for these. I mentioned this on the SC PR too but you probably didn't see [my comment](https://github.com/google/site-kit-wp/pull/2634#discussion_r568844042) since it was resolved shortly after 😄